worlds/lamulana/RcdMod.py

```diff
@@ -2,515 +2,516 @@
+
+        surface_scan_flag = GLOBAL_FLAGS["replacement_surface_map_scan"]
+        scan.test_operations[0].flag = surface_scan_flag
+        scan.write_operations[0].flag = surface_scan_flag
+        location.test_operations[0].flag = surface_scan_flag
 
-    self.__add_operation_to_object("write", location, surface_scan_flag, WRITE_OPERATIONS["add"], 1)
+        self.__add_operation_to_object("write", location, surface_scan_flag, WRITE_OPERATIONS["add"], 1)
 
 
-  def __give_starting_items(self, items) -> None:
-    flag_counter = 0
-    starting_room = self.file_contents.zones[1].rooms[2].screens[1]
+    def __give_starting_items(self, items) -> None:
+        flag_counter = 0
+        starting_room = self.file_contents.zones[1].rooms[2].screens[1]
 
-    for item_name in items:
-      item = item_table[item_name]
+        for item_name in items:
+            item = item_table[item_name]
 
-      item_giver = InstantItem(x=0, y=0, item=item.game_code, width=160, height=120, sound=39)
-      test_ops = [Operation.create(GLOBAL_FLAGS["starting_items"], TEST_OPERATIONS["eq"], flag_counter)]
-      write_ops = [
-        Operation.create(GLOBAL_FLAGS["starting_items"], WRITE_OPERATIONS["add"], 1),
-        Operation.create(item.obtain_flag, WRITE_OPERATIONS["add"], item.obtain_value)
-      ]
-      item_giver.add_ops(test_ops, write_ops)
-      item_giver.add_to_screen(self, starting_room)
+            item_giver = InstantItem(x=0, y=0, item=item.game_code, width=160, height=120, sound=39)
+            test_ops = [Operation.create(GLOBAL_FLAGS["starting_items"], TEST_OPERATIONS["eq"], flag_counter)]
+            write_ops = [
+                Operation.create(GLOBAL_FLAGS["starting_items"], WRITE_OPERATIONS["add"], 1),
+                Operation.create(item.obtain_flag, WRITE_OPERATIONS["add"], item.obtain_value)
+            ]
+            item_giver.add_ops(test_ops, write_ops)
+            item_giver.add_to_screen(self, starting_room)
 
-      flag_counter += 1
+            flag_counter += 1
 
 
-  def __rewrite_diary_chest(self) -> None:
-    diary_location = next((location for _, location in enumerate(get_locations_by_region(None, None, None)["Shrine of the Mother [Main]"]) if location.name == "Shrine of the Mother - Diary Chest"), None)
-    for zone_index in diary_location.zones:
-      diary_screen = self.file_contents.zones[zone_index].rooms[diary_location.room].screens[diary_location.screen]
-      diary_chest = next((o for _,o in enumerate(diary_screen.objects_with_position)
-        if o.id == RCD_OBJECTS["chest"]), None)
+    def __rewrite_diary_chest(self) -> None:
+        diary_location = LocationData("Shrine of the Mother - Diary Chest", 2359051, is_cursable=True, file_type='rcd', zones=[9], room=2, screen=1, object_type=0x2c, item_id=72, obtain_flag=0x104, obtain_value=2)
```

Comment:
This defeats the purpose of what was being done. The goal was to retrieve the necessary data (zone, room, screen) based off of the name key alone, so that those values aren't duplicated, so this copy/paste actually significantly increases the amount of duplication.

That being said, this entire approach is no longer how I'm handling RCD modifications and I had missed this line in some refactors I did so I don't think it's very consequential. I'll make a separate update in the future to rework this altogether

Reply:
I had done this to remove the number of calls using the weird `world = None` workaround instead of just having a separate function, but I still had to use `None` anyway for the name_to_id lookup. Reverted back to the previous style